### ISSUE TITLE
[BUG] Solution build does not leverage tasks.json

### ISSUE THREAD

**Issue by jaredpar** [microsoft/vscode-dotnettools#1346]
### Describe the Issue

The solution build experience uses `dotnet build` instead of using build items defined in `tasks.json`. That means all of the effort that we've gone through to customize the build for VS Code, and make it reliable, does not work when using DevKit. 

This causes downstream effects. Consider #1343 where problems with test window are exacerbated because it is overbuilding on Linux. This is not a problem for any other aspect of our build because tasks.json allows us to handle Linux vs. Windows gracefully. 

### Steps To Reproduce

1. Open a solution
2. Right click on solution explorer and select _Build_
3. 
![image](https://github.com/user-attachments/assets/a3d389a3-280d-4533-b4b1-c1c69d6414df)

### Expected Behavior

This action uses one of the available build tasks in my tasks.json. Or at least it should be configurable to do so.

**Comment by simonlampen** [comment]
I agree, having the ability to override the context menu on a project by project basis using something like tasks.json would be very useful. Also allowing nested tasks.json in project subfolders would be very useful. This way each project could have build overrides.

**Comment by sebwills** [comment]
+1. This would presumably resolve these too:
#1362 
#1363 
#1364
